core/app/models/concerns/spree/user_address_book.rb

```diff
@@ -3,7 +3,25 @@ module UserAddressBook
     extend ActiveSupport::Concern
 
     included do
-      has_many :user_addresses, foreign_key: "user_id", class_name: "Spree::UserAddress"
+      has_many :user_addresses, foreign_key: "user_id", class_name: "Spree::UserAddress" do
+
+        def find_first_by_address_values(address_attrs)
+          sql_record = with_address_values(address_attrs).first
+          return nil unless sql_record.present?
+          detect { |ua| ua.id == sql_record.id } #ensures we use a cached version
```

Comment:
Would it be possible to do the entire match here, instead of going to SQL and then finding the same in memory object?

Maybe something like `detect{ |ua| ua.address_values == address_attrs }`